inbm-lib/inbm_common_lib/validater.py

```diff
@@ -36,6 +36,21 @@ def validate_date(date: str) -> str:
     except ValueError:
         raise argparse.ArgumentTypeError(f"Not a valid date - format YYYY-MM-DD: '{date}")
 
+def validate_guid(value: str, param_type: str, max_size: int) -> str:
+    """Validates that the user inputted string does not exceed the maximum allowed
+        @param value: string entered by user
+        @param param_type: parameter type
+        @param max_size: maximum size allowed for the string
+        @return: entered string if it passes the length check
+        @raise argparse.ArgumentTypeError: Invalid guid format
+        """
+    if not bool(re.match("^[A-Fa-f0-9-]*$", str(value))):
+        raise argparse.ArgumentTypeError(f"guid does not support characters other than hexdigits and dash")
+
+    if len(value) > max_size:
```

Comment:
Using the regex "^[{]?[0-9a-fA-F]{8}" + "-([0-9a-fA-F]{4}-)" + "{3}[0-9a-fA-F]{12}[}]?$"

you can remove the length check.  And won't need the param_type and max_size parameters to this method.